### DISCUSSION TITLE
Type error, when using useFieldArray

### DISCUSSION THREAD

**Discussion by josias-r** [react-hook-form/react-hook-form#5585] [Q&A] [answered]
**Describe the bug**
If I use Controller with useFieldArray I get a strange type behavior when iterating through the fields. When I use the index (number) from the Array.map, the Controller interface breaks and throws me a type error (for the defaultValue prop for example) for some reason.
If I use a specific number though, then it works. So it only happens with type "number"

**To Reproduce**
V7 useFieldArray with typescript (see codesandbox)

**Codesandbox link (Required)**
https://codesandbox.io/s/nostalgic-tu-8of4j?file=/src/Playground.tsx:648-684
**Expected behavior**
No type error :)

**Comment by bluebill1049** [comment, marked as answer]
<img width="862" alt="Screen Shot 2021-06-12 at 10 44 36 pm" src="https://user-images.githubusercontent.com/10513364/121776363-c513a580-cbcf-11eb-8249-c2b021742eba.png">

there is TS limitations.

**Reply by josias-r** [reply]
Interesting and unfortunate :/ , I will go with this (for the name prop of Controller), this works too:
````
${foooo}.${i as 0}.name\` as const

**Reply by josias-r** [reply]
Thank you very much for the quick reply, you saved me some hours of additional trial and error :P

**Reply by bluebill1049** [reply]
no worries @josias-r yea, there r some limitations with the usage of TS. unfortunately :(
